crates/test-programs/wasi-tests/src/bin/path_open_read_without_rights.rs

```diff
@@ -0,0 +1,96 @@
+use std::{env, process};
+use wasi_old::wasi_unstable;
```

Comment:
Should we update this to use the `wasi` crate given it's a new test program?

Reply:
I'm quite confused how to cope with the snapshot stuff or `wasi` vs `wasi_old`, so any input is appreciated.

Reply:
As an example, the test program added for #711 uses the `wasi` crate, which will currently target the preview1 impl.

I assume we'll be moving all the test programs over to the `wasi` crate sooner or later.

Additionally, I think we'll want dedicated tests targeted at previous previews for the sole purpose of testing backwards compatibility for WASI interface changes so we don't have to snapshot *all* of the test programs themselves for every snapshot of WASI, but perhaps @sunfishcode @alexcrichton @kubkon know more regarding that.

Reply:
Yep, the current trend is to use `wasi` in tests as @peterhuene suggested. As far as I remember, @alexcrichton has already started porting some of the tests to using `wasi` instead of `wasi_old`.

In the future, I definitely agree we should have a way to test different versions of the snapshot, preferably in a way that's automatic.